app/routes/app/brainstorm/new.tsx

```diff
@@ -21,37 +26,95 @@ export const loader = async () => {
 
 const validator = withZod(LaborMarketNewSchema);
 
-export const action = async ({ request }: DataFunctionArgs) => {
-  const data = await validator.validate(await request.formData());
-  if (data.error) return validationError(data.error);
+type ActionResponse = { preparedLaborMarket: LaborMarketPrepared } | ValidationErrorResponseData;
+export const action = async ({ request }: ActionArgs) => {
+  const result = await validator.validate(await request.formData());
+  if (result.error) return validationError(result.error);
+
+  // Additional server side validation
+  if (result.data.launchAccess === "delegates") {
+    const badgerDataMissing = !result.data.launchBadgerAddress || !result.data.launchBadgerTokenId;
+    if (badgerDataMissing) {
+      return validationError({
+        fieldErrors: {
+          launchBadgerAddress: !result.data.launchBadgerAddress
+            ? "Required if delegates selected for launch access"
+            : "",
+          launchBadgerTokenId: !result.data.launchBadgerTokenId
+            ? "Required if delegates selected for launch access"
+            : "",
+        },
+      });
+    }

```

Comment:
Maybe nicer to add this validation directly to the LaborMarketNewSchema. Something like:
```typescript
export const LaborMarketNewSchema = LaborMarketSchema.omit({
  address: true,
  sponsorAddress: true,
}).superRefine((data, ctx) => {
  if (data.launchAccess !== "delegates") return true;
  if (!data.launchBadgerAddress || data.launchBadgerAddress.length === 0) {
    ctx.addIssue({
      path: ["launchBadgerAddress"],
      code: z.ZodIssueCode.too_small,
      minimum: 1,
      type: "string",
      inclusive: true,
      message: "Required if delegates selected for launch access",
    });
  }
  if (!data.launchBadgerAddress || data.launchBadgerAddress.length === 0) {
    ctx.addIssue({
      path: ["launchBadgerTokenId"],
      code: z.ZodIssueCode.too_small,
      minimum: 1,
      type: "string",
      inclusive: true,
      message: "Required if delegates selected for launch access",
    });
  }
});
```